src/main/java/org/logstash/beats/BeatsHandler.java

```diff
@@ -133,4 +133,13 @@ private String format(String message) {
 
         return "[local: " + localhost + ", remote: " + remotehost + "] " + message;
     }
+
+    private static final int MAX_CAUSE_NESTING = 10;
+
+    private static Throwable extractCause(final Throwable ex, final int nesting) {
```

Comment:
Could also use a slow "pointer"/fast "pointer" to detect exception loops, if an exception loop is common for this kind of error, but this is ok too if not

Reply:
have checked this manually enforcing a TLS certificate validation failure - which gets us 4 levels of nested causes.
at this point I am not sure loops are common just wanted to make sure we do not accidentally end up looping too much here, esp. as these are considered non-fatal and won't halt plugin execution.

imho, what's important is that we try the extraction since the info logging the plugin did previously was not providing useful information on the 'real' cause + also gave 'unclear' logging based on the following condition:

`cause.getMessage() == null ? cause.getClass().toString() : cause.getMessage()`
which made me previously believe we had a `javax.net.ssl.SSLHandshakeException: General SSLEngine problem` around here while in fact it was wrapped in a `io.netty.handler.codec.DecoderException` with `"javax.net.ssl.SSLHandshakeException: General SSLEngine problem"` as the message.

the whole exception stack is later printed by Netty's logger.

Reply:
Makes sense - I only asked about Exception loops because I encountered them working on the JMS plugin, and added a guard against them.